index.js

```diff
@@ -17,7 +19,13 @@ module.exports = function (str, filepath) {
 
 	return pify(mkdirp, Promise)(path.dirname(fullpath))
 		.then(function () {
-			return pify(fs.writeFile, Promise)(fullpath, str);
+			return isStream(str) ?
+				new Promise((resolve, reject) => {
+					str.pipe(fs.createWriteStream(fullpath))
```

Comment:
Okay, I just played around with a bunch of different patterns and ran into an appalling number of cases where despite `error` events being fired, data would still end up on disk. E.g. because `error` events don't end the stream, it's easy for data to keep flowing after an error. I don't think this module should allow that.

See my latest commit which includes this new pattern:

```js
function pipeStream() {
    return new Promise(function (resolve, reject) {
        var writable = fs.createWriteStream(fullpath)
        str
            .on('error', function (err) {
                // Make sure to reject before writable.end(), otherwise the
                // 'finish' event will fire and we will resolve rather than
                // reject the promise.
                reject(err);
                str.unpipe(writable);
                writable.end();
            })
            .pipe(writable)
            .on('error', reject)
            .on('finish', resolve);
    });
}
```